scripts/build-docs.ts

```diff
@@ -0,0 +1,115 @@
+import * as path from "path";
+import * as fs from "fs-extra";
+import { Application, TSConfigReader, TypeDocOptions } from "typedoc";
+
+const OUTPUT_DIR = "dist/docs";
+const INDEX_FILE = "doc/index.md";
+
+const GITHUB_FORMAT = 'github';
+const PC_DEV_CENTER_FORMAT = 'purecloudDevCenter';
+const SUPPORTED_DOC_OUTPUT_FORMATS = [GITHUB_FORMAT, PC_DEV_CENTER_FORMAT];
+let docMdOutputFormat = process.env.DOC_MD_OUTPUT_FORMAT || PC_DEV_CENTER_FORMAT;
+if (SUPPORTED_DOC_OUTPUT_FORMATS.indexOf(docMdOutputFormat) < 0) {
+    console.error(`Unknown MD Output Format Specified: '${docMdOutputFormat}'`);
+    process.exit(1);
+}
+
+let flagDocsAsPreview = true;
+if (process.env.FLAG_DOCS_AS_PREVIEW) {
+    ['false', 'f', '0'].forEach(currFalseValue => {
+        if (process.env.FLAG_DOCS_AS_PREVIEW === currFalseValue) {
+            flagDocsAsPreview = false;
+        }
+    });
+}
+
+// Options added by the Typedoc Markdown Plugin
+interface MarkdownPluginOptions {
+    skipSidebar: boolean;
+    hideBreadcrumbs: boolean;
+}
+
+type DocsConfig = Partial<TypeDocOptions & MarkdownPluginOptions>;
+
+const docsConfig: DocsConfig = {
+    mode: "file",
+    name: "Client App SDK",
+    logger: 'none',
+    theme: "docusaurus", // Theme that generates output closest to Dev Center format
+    skipSidebar: true, // Disable sidebar generation
+    disableSources: true, // Don't show the file for each method/class
+    hideBreadcrumbs: true, // Dev Center handles breadcrumbs for us
+    readme: "none",
+    plugin: [
+        'typedoc-plugin-markdown',
+        'typedoc-plugin-no-inherit'
+    ],
+    exclude: [
+        "**/utils/**",
+        "**/.babelrc",
+        "**/.eslintrc.js",
+        "**/*+(Spec).ts"
+    ],
+    excludeExternals: true,
+    excludePrivate: true,
+    excludeProtected: true,
+    excludeNotExported: true,
+    excludeNotDocumented: true
+};
+
+const transformLinks = (buffer: string, ext: string) => {
+    return buffer.replace(/(\[[^\]]+\][^\)]+)(\.md)(\)|\#[^\)]*\))/gm, `$1${ext}$3`);
+};
+
+const prependDocHeaderAttribute = (buffer: string, attr: string) => {
```

Comment:
Yeah, the generated doc always spits out a header with some predefined fields, but I think its probably a good sanity check since we do have one markdown file that is just copied and not generated.